### ISSUE TITLE
Shared calendar permission

### ISSUE THREAD

**Issue by ITTRI4710** [owncloud/calendar#882]
Creating, modifying and deleting dropdown permission is missing. See Image.
![2018-02-07 12_58_14-microsoft edge](https://user-images.githubusercontent.com/34244878/36094135-5b3060f0-0fed-11e8-8a00-479931ff93b6.png)

ownCloud 10.0.3 (stable)
Calendar 1.5.4

**Comment by DeepDiver1975** [comment]
If I'm not miss taken detailed sharing permissions never got implemented in the calendar app.

**Comment by ITTRI4710** [comment]
Before we upgrade to the new version, we could additionally select.

**Comment by DeepDiver1975** [comment]
THX - we will look into this 

@Julian1998 - one for you - THX

**Comment by ittrend4710** [comment]
i'm watching ;-)

**Comment by Julian1998** [comment]
Weird... @ITTRI4710 Which version did you use before upgrading?

Not sure if this is a bug... I couldn't even find the template for detailed sharing permissions

**Comment by ittrend4710** [comment]
@Julian1998 

ownCloud 8.1.9
Update ownCloud -> 8.2.x
Update ownCloud 8.2.x -> 9.0.x
Update 9.0 -> 9.1 
Update 9.1 -> 10.x

**Comment by Julian1998** [comment]
@ittrend4710 thx! I will have a look the next few days... Btw why are you talking for @ITTRI4710 ;) Your account?

**Comment by ittrend4710** [comment]
Hello there :-)
just to be clear:
ITTRI4710 -> is one of our support engineers
ITTTREND4710 -> is me ... sysadmin
2 different persons working on the same issue...
Best,
Mike

**Comment by Julian1998** [comment]
Ok! Seems like detailed actions are never used since version 1.0 so I would not call it bug...

We could implement detailed sharing actions but I don't see a need for this. This needs to be more discussed!

**Comment by ittrend4710** [comment]
I don't know if anyone used it in the versions between. Our customer used it in the old version and is missing it in the new one...

An implement would be fine!

Best,
Mike

**Comment by DeepDiver1975** [comment]
Ah .. so the more detailed sharing permissions are from the very first implementation?

Generally speaking we need to move to a much simpler permission schema:
read and read/write is enough.

Background:
As soon as you have write permission you can delete a entity by wiping all it's contents.

What would be a reasonable scenario where one wants to pass a different set of permissions? Create but not modify or delete? Hmmm ....

**Comment by ittrend4710** [comment]
secretary can arrange new appointments if there is free time for it, but cann't edit or delete existing appointments because they are managed by another person?

**Comment by DeepDiver1975** [comment]
> secretary can arrange new appointments if there is free time for it, but cann't edit or delete existing appointments because they are managed by another person?

well - how reasonable is a create permission without an update permission? Assume the secretary makes a mistake while creating. 

A much better workflow would be to give read only permission to the secretary and use explicit the attendee feature of caldav where the user can then accept or reject the event.
Accepting will write the event to the shared calendar.

Or even use free-busy feature of caldav - no need to share the calendar then at all.

**Comment by ittrend4710** [comment]
-->  It’s a very special usecase … headquarter with many national organizations All I can tell is that customer is missing a feature from previous version … they even asked me for a drowngrade (which I’ll not do…) Best,Mike

**Comment by DeepDiver1975** [comment]
@ITTRI4710 Can you please reply with text only emails? This is a mess ;-)

**Comment by DeepDiver1975** [comment]
> --> It’s a very special usecase … headquarter with many national organizations All I can tell is that customer is missing a feature from previous version … they even asked me for a drowngrade (which I’ll not do…) Best,Mike

I for sure understand that customers a crying for features especially those we removed on purpose ;-)

All I'm proposing is to reevaluate the process your customers are using. Furthermore I try to understand the usage scenarios.

THX a lot

**Comment by ittrend4710** [comment]
Text only ;-)

We'll ask our customer for the detailled usecase and inform you afterwards...

Best,
Mike

**Comment by ITTRI4710** [comment]
Hi!

The customer wants to prevent appointments that those who can change the appointment but can not delete it. So that it can not be accidentally deleted and that only the calendar owner can do.

Best,
Roland

**Comment by ITTRI4710** [comment]
Hi!

...any news about that?

best,
Roland

**Comment by Julian1998** [comment]
Hi, I am sorry but you have to be patient... There is some stuff that has to be done first :)
Btw this is open source... so don't be shy and try to fix it. We will help as good as we can :)

**Comment by ITTRI4710** [comment]
Hi!

Can you tell me if you already had time for it?

best,
Roland

**Comment by Julian1998** [comment]
Hi! Unluckily I had not time for this yet. Furthermore I am still very unsure about the advantage of this feature. There is already a feature that allows you to avoid the editing of a shared calendar. I know that this is needed for you but as long as just you are crying for this feature we have to postpone it.... 

This doesn't mean we won't implement it in the future but we need to discuss this feature a little bit more... (There is a reason why we removed that options)

Best, Julian

**Comment by ITTRI4710** [comment]
Hi!

Did you already talk about it?

best,
Roland

**Comment by ITTRI4710** [comment]
Hi!
Can you tell me if you already had time for it?
best,
Roland

**Comment by vierundvierzig** [comment]
Hello!

I am very interested in this feature as well! We use Calendar on owncloud in our organisation to book some devices for use. We have 5 different calendars, admin is owner of all of them. 10 persons should have an access for calendars, they should be able to add, modify or delete events but they cannot have option to delete whole calendar - then we permanently lost our schedule.

Could you tell me if you are still considering adding this feature?

Best regards,
Adam

**Comment by Julian1998** [comment]
Well... as this is still a community driven project. We need someone who develops this. We are open for any features/bugfixes by anyone. I would love to see this feature. But I don't have time to do this on my own for now :/. So if anyone here would love to see this as well - go for it. Otherwise you/we need to wait until I got time (and also fun) for sth like this.

**Comment by vierundvierzig** [comment]
Sure, I understand that. But I'm not a coder so I count on you and that's all I can do about it. You said that ITTRI is only one person interested in this feature so it is my voice that I'm the another one ;)

**Comment by ITTRI4710** [comment]
This feature was available with version 8 but with 10 it was gone

**Comment by Julian1998** [comment]
@ITTRI4710 The problem is that calendar 10 (if the versions u said are correctly) is a complete rework of the old calendar. (That's why the calendar that you refer to are in the owncloud archive)

This being said, you are probably right that this feature worked before, but just copying and pasting won't work as this are two complete different frameworks/apps.

So this would take it's time

**Comment by Julian1998** [comment]
Edit: There is the proof ;) https://github.com/owncloudarchive/calendar
